### DISCUSSION TITLE
Feature request: ability to disable preview while exporting folders to speed up the process

### DISCUSSION THREAD

**Discussion by M-U-X** [4sval/FModel#490] [Q&A]
in the folder list on the left side, next to archive, if you click a folder depending on the existing assets size, and number, and decide to export all of said folder's hierarchy's models, or animations, it can take ages, probably a huge part of the process is time wasted on displaying uassets in the viewer, and it seems to slow down with time, so there might be a memory leak or a resource exhaustion going on, a simple switch in the sitting to disable preview of uassets files while mass exporting could be extremely helpful.

**Comment by 4sval** [comment]
preview is not triggered on those conditions

**Reply by M-U-X** [reply]
not the 3d preview, but it keep previewing the uassets file itself in the "code/editor" preview window or whatever it's called
![image](https://github.com/user-attachments/assets/8e307034-4232-4076-aa5b-b71d05789dc1)

**Reply by 4sval** [reply]
that’s the whole point of the program
if you don’t need that and only batch export stuff please use https://github.com/FabianFG/CUE4Parse

**Reply by M-U-X** [reply]
I don't understand, how would adding an option/toggle that could speed mass export -which is available in Fmodel- defeat the whole point of the program, it'd simply speed the export for people who want to? and don't want to use a library instead of a GUI

**Reply by 4sval** [reply]
disabling the json preview will not speed up the export
most of the time is spent building the mesh to export and saving textures in materials

if you want it to be faster use the UEFormat model export format and disable "Save Materials Embedded within Meshes"

**Reply by M-U-X** [reply]
something is actually wrong with it like I said, it doesn't seem to be the preview, but when exporting a big enough number of objects it slows to a halt almost, I've a 14700k cpu, & 32GB of ram, so it's not the pc, steps to consistently reproduce on my end:

for the slow down:
1-valorant-> load all -> folders (shooter, content, equippables, and export the animations of a large subfolder, for example melee or AK, after a while the export will slow down so much, & if you clear the log it starts going at normal seed again.

for crash:
1- same thing but export the animation of the equippables bomb folder, always causes a crash on my end

**Reply by 4sval** [reply]
this has been [fixed](https://github.com/4sval/FModel/commit/fabc064a4618b7bd9ab3b7a9d3ffe58f93ab10e1) in the latest QA Testing update but please reconsider your workflow.

FModel doesn't have a scan feature yet, so atm, you're asking FModel to parse nearly 90k packages for only ~6k animations. By parsing all these useless packages FModel will encounter broken ones that will cause a hard crash. There's nothing we can do about it, we don't own the data we parse.

Valorant is well made and use a comprehensive folder structure, use it to your advantage and bulk export stuff via the search window (update to QA first).
If you blindly need everything, again, please use https://github.com/FabianFG/CUE4Parse
